packages/ui/src/components/DatePicker/Calendar/Calendar.tsx

```diff
@@ -1,5 +1,7 @@
 import React from 'react'
-import { Box, Stack, Text } from '@highlight-run/ui'
+import { Box } from '../../Box/Box'
+import { Stack } from '../../Stack/Stack'
+import { Text } from '../../Text/Text'
```

Comment:
Had to do a bunch of untangling of circular dependencies in this commit.

Reflame's development mode makes use of [es-module-shims](https://github.com/guybedford/es-module-shims)' shim mode to be able to hook into module loading and update import maps at runtime for its React Fast Refresh implementation. es-module-shims generally does a good job of supporting circular imports transparently, but it seems to be running into some trouble with this specific pattern where index.ts is full of reexports. I'll try to find a minimal repro and file an issue there.

In the longer term, it may be worth considering how we could avoid circular dependencies as much as possible regardless, since they are a blocker to effective code splitting in bundled workflows. Reflame can probably help with this by warning on any circular deps it encounters as part of its dependency analysis.

Reply:
nice! yeah as i mentioned in the other comment, this will be much more manageable if someone introduces a PR that breaks this but the reflame preview fails (and hence notifies the author)

Reply:
This particular case is a bug on Reflame's side (or rather in es-module-shim's shim mode), since circular dependencies are supported just fine in standard modules (i.e. they work for production mode where we don't use es-module-shim's shim mode).

But yeah we'll definitely add warnings for circular dependencies sometime down the line since they cause plenty of other nasty problems (most notably for code-splitting since we can't split off anything within a cycle).